### PR TITLE
Revert "overlay: remove /boot/ignition on upgrade if present"

### LINKED ISSUE #1094: Drop `coreos-cleanup-ignition-config.service` after the next barrier release on all streams
Ignition will now automatically remove `/boot/ignition` on first boot, so `coreos-cleanup-ignition-config.service` only needs to handle old nodes.  Once all of those have passed through a barrier, we can revert #1093.

For https://github.com/coreos/fedora-coreos-tracker/issues/889.

### PR DESCRIPTION
**Pull request by bgilbert**
This was a fixup for existing nodes on upgrade.  Now that we've passed through the F35 barrier release, drop the fixup.

Fixes #1094.  Reverts #1093.  xref https://github.com/coreos/fedora-coreos-tracker/issues/884.

### CODE REVIEW

**Comment by bgilbert** [comment]
Marking draft until we land F35 in stable.  Otherwise, if we ended up skipping a stable promotion due to bugs, we'd risk adding the barrier after this unit was already removed.

**Comment by bgilbert** [comment]
Rebased and ready for review!

File: tests/kola/migration/installer-cleanup [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-case "${AUTOPKGTEST_REBOOT_MARK:-}" in
-"")
-    sudo mount -o remount,rw /boot
-    sudo mkdir -p /boot/ignition
-    sudo touch /boot/ignition/config.ign
-    /tmp/autopkgtest-reboot rebooted
-    ;;
-rebooted)
-    [[ ! -e /boot/ignition ]]
-    ok "/boot/ignition was removed"
```
Comment (jlebon) [inline tests/kola/migration/installer-cleanup:33]: For posterity, note `kola testiso` does verify that `coreos-ignition-firstboot-complete.service` nukes `/boot/ignition` (see https://github.com/coreos/coreos-assembler/pull/2283).

**Review by jlebon — APPROVED** [review]

**Review by dustymabe — APPROVED** [review]
LGTM

**Pull request merged**